### PR TITLE
rules: Add rules for amd-pmf

### PR DESCRIPTION
**Pull request by superm1**
The AMD-PMF driver can emit some system events that indicate that userspace should take an action such as locking the machine, suspending, or hibernating.

Wire up udev rules to these events as suggested in the kernel documentation.

Link: https://github.com/torvalds/linux/blob/v6.8-rc1/Documentation/admin-guide/pmf.rst

### CODE REVIEW

File: rules.d/70-amd-pmf.rules.in [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# do not edit this file, it will be overwritten on update
+
+ACTION!="change", GOTO="pmf_end"
+DRIVERS!="amd-pmf", GOTO="pmf_end"
+
+ENV{EVENT_ID}=="0", RUN+="{{BINDIR}}/systemctl suspend"
```
Comment (YHNdnzj) [inline rules.d/70-amd-pmf.rules.in]: This is just wrong. Please handle these in logind... logind might not be running when udev tries to handle these events, also the actual action to perform should be controllable through logind.conf like power buttons/lid switch.

**Review by YHNdnzj — CHANGES_REQUESTED** [review]

File: rules.d/70-amd-pmf.rules.in (hunk 1 above)
Comment (superm1) [reply rules.d/70-amd-pmf.rules.in]: OK.  Let me just do the lock-sessions rule for now then, and the sleep/hibernate will need to get wired up as you suggest.

Comment (superm1) [reply rules.d/70-amd-pmf.rules.in]: (and there is previous precedent for `lock-sessions` - see *71-seat.rules.in*)

**Comment by poettering** [comment]
i have no idea what amd-pmf really is, the docs are a bit useless on that. but why doesn't this just generate input events? i.e. KEY_SUSPEND, KEY_SCREENLOCK and so on?

We listen to KEY_SUSPEND anyway, it would be trivial then.

But anyway, I really don't grok what this actually is: when precisely are these events sent? For what kind of devices is this, can you explain?

**Comment by superm1** [comment]
> i have no idea what amd-pmf really is, the docs are a bit useless on that. but why doesn't this just generate input events? i.e. KEY_SUSPEND, KEY_SCREENLOCK and so on?
> 
> We listen to KEY_SUSPEND anyway, it would be trivial then.
> 

That's a great suggestion thanks.

**Pull request closed without merging**